src/products/src/products-service/repository.go

```diff
@@ -373,3 +373,75 @@ func RepoFindALLProduct() Products {
 
 	return f
 }
+
+func RepoUpdateProduct(product Product) Product {
+
+	log.Println("UpdateProducts for id, new quantity: ", product)
+
+	av, err := dynamodbattribute.MarshalMap(product)
+
+	if err != nil {
+		fmt.Println("Got error calling dynamodbattribute MarshalMap:")
+		fmt.Println(err.Error())
+		return product
+	}
+
+	input := &dynamodb.PutItemInput{
+		Item:      av,
+		TableName: aws.String(ddb_table_products),
+	}
+
+	_, err = dynamoclient.PutItem(input)
+	if err != nil {
+		fmt.Println("Got error calling PutItem:")
+		fmt.Println(err.Error())
+
+	}
+
+	return product
+
+}
+
+func RepoUpdateInventory(inventory Inventory) Product {
```

Comment:
Would an AtomicCounter operation be more appropriate here to avoid concurrent update issues?

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.AtomicCounters

Reply:
I guess you meant Conditional Writes ? https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.ConditionalUpdate

I will try to implement it

Reply:
> I guess you meant Conditional Writes ? 

It depends on the semantics of the API. If the API is intended to be, "increase/decrease inventory by N", then an atomic counter seems appropriate. If the API is intended to be, "change/set the inventory to N", then a conditional write seems appropriate. 

I could see use-cases for both, actually. When completing an order during checkout, we're looking to decrement the inventory. When processing a return or voiding an order, we're looking to increment the inventory. The current value doesn't matter (except, I suppose, when trying to decrement when the inventory is zero).

When performing an inventory audit where you're counting inventory, we're looking to update/set the inventory value.